### PR TITLE
Password protection for 'public' websites

### PR DESCRIPTION
**Pull request by yevheniikulisidi**
https://github.com/Swetrix/roadmap/issues/142

### CODE REVIEW

File: src/project/project.controller.ts [hunk 1]
```diff
@@ -1351,13 +1357,23 @@ export class ProjectController {
       project.name = projectDTO.name
       project.public = projectDTO.public
 
+      if (project.public && projectDTO.password) {
```
Comment (kruzhambus) [inline src/project/project.controller.ts:1360]: we use public or protected project. delete project.public from if

File: src/project/project.controller.ts [hunk 2]
```diff
@@ -1351,13 +1357,23 @@ export class ProjectController {
...
+        project.passwordHash = await hash(projectDTO.password, 10)
+        project.isPasswordProtected = true
+      }
+
+      if (project.isPasswordProtected && projectDTO.password === null) {
```
Comment (kruzhambus) [inline src/project/project.controller.ts:1365]: add to projectDTO value isPasswordProtected and when the user off passwordHash, you get value isPasswordProtected: false when on you get isPasswordProtected: true and passwordHash

Comment (kruzhambus) [reply src/project/project.controller.ts:1365]: so you need to delete this if

**Comment by kruzhambus** [comment]
add replace password to passwordHash, because password is not comfortable in fe

**Comment by Blaumaus** [comment]
Closed in favour of https://github.com/Swetrix/swetrix-api/pull/148

**Pull request closed without merging**